bucket/quickviewer.json

```
@@ -0,0 +1,55 @@
+{
+    "version": "1.1.6",
+    "homepage": "https://kanryu.github.io/quickviewer/",
+    "description": "An image/comic viewer.",
+    "license": "GPL-3.0-only",
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/kanryu/quickviewer/releases/download/v1.1.6/QuickViewer-portable-1.1.6-x64.zip",
+            "hash": "2d8cd19f98e93c88f949041120f86f6973e20beb67322ca1554ba206af406234",
+            "extract_dir": "QuickViewer-portable-1.1.6-x64"
+        },
+        "32bit": {
+            "url": "https://github.com/kanryu/quickviewer/releases/download/v1.1.6/QuickViewer-portable-1.1.6-x86.zip",
+            "hash": "a415cf4d27e22a544a2ae8b4a781dfc50bd37aa6d8b6e769997a4f938d3775e9",
+            "extract_dir": "QuickViewer-portable-1.1.6-x86"
+        }
+    },
+    "installer": {
+        "script": [
+            "foreach ($file in @('progress.ini', 'quickviewer.ini')) {",
+            "    if (!(Test-Path \"$persist_dir\\$file\")) {",
+            "        New-Item \"$persist_dir\\$file\" -Force | Out-Null",
+            "    }",
+            "    Copy-Item \"$persist_dir\\$file\" \"$dir\\$file\" -Force",
+            "}"
+        ]
+    },
+    "uninstaller": {
+        "script": [
```

I forgot force, sorry..
```suggestion
        "script": "Copy-Item \"$dir\\*\" \"$persist_dir\" -Include 'progress.ini', 'quickviewer.ini' -Force"
```

---

How about `"Copy-Item \"$dir\\*.ini\" \"$persist_dir\" -Force"` ? And it works well without force.

---

I do not know what ini files are there. If you want to persist only these two, then just copy these two.

---

There is no other ini file. We can simplify it as we do not need to specify which ini file should be persist.

---

There is nothing which could be earned when copying all ini files. Yeah funciton will be shorter, but that is all.

---

I wonder will this make the installer more clear?

```
    "installer": {
        "script": [
            "if (!(Test-Path \"$persist_dir\")) { New-Item \"$persist_dir\" -ItemType Directory -Force | Out-Null }",
            "Copy-Item \"$persist_dir\\*.ini\" \"$dir\" -Force"
        ]
    },
    "uninstaller": {
        "script": "Copy-Item \"$dir\\*.ini\" \"$persist_dir\" -Force"
    },
```

---

How will this solve that these configuration files do not exists in archive?

---

Keep it as is now.

---

They will be created when first run. So we can persist them only when they exist. We can also do this on those file which can't be created manully. Such as

```
            "    if (Test-Path \"$persist_dir\\$file\") {",
            "        Copy-Item \"$persist_dir\\$file\" \"$dir\\$file\" -Force",
            "    }",
```

---

These special use cases will be covered in lukesampson/scoop#3212

Until persistence is reworked keep it as simple and readable as you can.

---

Great! Thanks!